lib/options.nix

```diff
@@ -99,6 +99,20 @@ rec {
     type = lib.types.bool;
   };
 
+  /* Creaties an Option attribute set for an option that specifies the
+     package a module should use.
+
+     Example:
+       mkPackageOption { name = "foo"; default = pkgs.hello; }
+       => { _type = "option"; default = «derivation /nix/store/3r2vg51hlxj3cx5vscp0vkv60bqxkaq0-hello-2.10.drv»; description = "The foo package to use."; example = «derivation /nix/store/3r2vg51hlxj3cx5vscp0vkv60bqxkaq0-hello-2.10.drv»; type = { ... }; }
+     */
+  mkPackageOption = { name, default, example ? default }:
+    mkOption {
+      type = lib.types.package;
+      description = "The ${name} package to use.";
+      inherit default example;
```

Comment:
this also needs a `defaultText` or the manual builds will fail, and the example should be a `literalExample` to match. at that point we'd have to specify the package not as a `pkgs.X.Y` attrpath, but pass `pkgs` and the attrpath into the function, the latter as a list of strings. at that point we're saving little over writing the option out completely.

(yes, this is not an ideal situation)

Reply:
Is there a lib function to navigate a path in an attrset?

Reply:
there's a few. `getAttrFromPath` is closest, but hard-aborts when it can't find the requested attribute (which is not ideal). `attrByPath` can be given a `(throw "attribute ${...} not found")` default instead of an `abort` default, so that would be better.

Reply:
Can you add `{ pkgs, lib }: ` to the start of the file to reference nixpkgs itself?

Reply:
I’ve made a new version, haven’t tested it though, since nixpkgs is so big and I don’t have a local copy (downloading ATM)